### PR TITLE
react_list-of-goods-js

### PR DESCRIPTION
**Pull request by AndriiYelieva**
https://AndriiYelieva.github.io/react_list-of-goods-js/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
...
+
+  const sortGoods = getPreparedGoods(goodsFromServer, sortList, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${cN({ 'is-light': sortList !== SORT_ASC })}`}
+          onClick={() => setSortList(SORT_ASC)}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:59]: it is bad practise to make arrow functions in jsx.
consider to make one funciton for each sortType which will be accept sortField and return arrow function.
```
const makeSetSortList = (field) => () => setSortList(field);
```
```suggestion
          onClick={(makeSetSortList(SORT_ASC)}
```

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
...
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${cN({ 'is-light': sortList !== SORT_BY_LENGTH })}`}
+          onClick={() => setSortList(SORT_BY_LENGTH)}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:67]:
```suggestion
          onClick={makeSetSortList(SORT_BY_LENGTH)}
```

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
...
+        >
+          Reverse
+        </button>
+
+        {(sortList || isReversed) && (
+          <button
+            onClick={() => {
+              setSortList('');
+              setIsReversed(false);
+            }}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:85]: make handler in component and pass reference here
```
const reset = () => {
  setSortList('');
  setIsReversed(false);
}
```
```suggestion
            onClick={reset}
```

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
...
+          className={`button is-success ${cN({ 'is-light': sortList !== SORT_BY_LENGTH })}`}
+          onClick={() => setSortList(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${cN({ 'is-light': !isReversed })}`}
+          onClick={() => setIsReversed(prevState => !prevState)}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:75]: same as for reset

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
...
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+//       <ul>
+//         {sortedGoods.map(good => <li data-cy="Good" key={good}>{good}</li>)}
+//       </ul>
+//     </div>
+//   );
+// };
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:171]: remove commented code

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**